### ISSUE TITLE
feat: Consider a portable / non install version

### ISSUE THREAD

**Issue by tomasts248** [janhq/jan#2643]
**Problem**
All the disadvantages of a regular installation and data portability caused by it.

**Success Criteria**
Being able to run Jan from a external drive and just doing a full back by copying the folder where it is extracted, using a portable installation is now a common good practice in software development and as such usually requested.

**Comment by Van-QA** [comment]
related to https://github.com/janhq/jan/issues/1435

**Comment by 0xSage** [comment]
Not likely to be fully supported

**We have some portability**, but I'm not sure 100% portability makes sense for us at the moment. 

Not sure if you disagree @dan-homebrew @hiento09 

e.g.
1. Executable vs user/app data: already portable. All app data is in ~/.jan 
2. Relative paths: requires rewriting a lot of things as we write to program files right now
3. Configuration files: already portable.

**Comment by hiento09** [comment]
@0xSage, I understand that `portable` here means the .exe file we ship to the user, and the user just needs to double-click for the Jan app to launch. It's similar to the `AppImage` format for Linux, where the user only needs to run `./jan.appImage` for the Jan app to start without needing to install it on the OS.

**Comment by hiento09** [comment]
I agree with you that our data is in ~/jan, and allowing the data folder to be changed, so this portable version doesn't make much sense to me at the moment as well.

**Comment by tomasts248** [comment]
Not sure here, but cant the actual program files folder be redirected? and just have a launcher that redirects from program files to maybe another "program files " but rather than in the OS, in the app folder

**Comment by v1ckxy** [comment]
> @0xSage, I understand that `portable` here means the .exe file we ship to the user, and the user just needs to double-click for the Jan app to launch. It's similar to the `AppImage` format for Linux, where the user only needs to run `./jan.appImage` for the Jan app to start without needing to install it on the OS.

Not exactly.
Portable means self-contained within the same path; just have a look at any application from portableapps.com:
There's a launcher, a data path and a binary path. But the same.

The point is not to throw a lot of crap around elsewhere, as typically any app does (starting from LM Studio, using not just one or two different paths, but three+squirrel temp files)

**Comment by LazyYuuki** [comment]
Closed as not planned or working on

**Comment by hellomynameisnone** [comment]
Hi I made a .bat file, is not perfect, but do the job, I hope someone more skilled can improve it

I recommend use "Sandboxie-Plus" and install it there, and move folder "Jan", "jan.ai.app", etc etc

Tested on v0.7.5

The bat file

```

@echo off
setlocal enabledelayedexpansion
title Jan Portable Launcher

echo ===============================================
echo        Starting Jan Portable (Self-Contained)
echo ===============================================
echo.

REM === Base path of the launcher ===
set "BASE=%~dp0"
set "PORTABLE=%BASE%AppData"

REM === Create portable structure if it doesn't exist ===
if not exist "%PORTABLE%\Local\Temp" mkdir "%PORTABLE%\Local\Temp"
if not exist "%PORTABLE%\Roaming" mkdir "%PORTABLE%\Roaming"
if not exist "%PORTABLE%\Local\Programs\Jan" mkdir "%PORTABLE%\Local\Programs\Jan"

REM === Critical portable environment variables ===
set "APPDATA=%PORTABLE%\Roaming"
set "LOCALAPPDATA=%PORTABLE%\Local"
set "USERPROFILE=%BASE%"
set "HOME=%BASE%"
set "TEMP=%PORTABLE%\Local\Temp"
set "TMP=%TEMP%"

REM === Executable path ===
set "JAN_EXE=%LOCALAPPDATA%\Programs\Jan\Jan.exe"

REM === Debug ===
echo [DEBUG] APPDATA=%APPDATA%
echo [DEBUG] LOCALAPPDATA=%LOCALAPPDATA%
echo [DEBUG] USERPROFILE=%USERPROFILE%
echo [DEBUG] HOME=%HOME%
echo [DEBUG] TEMP=%TEMP%
echo [DEBUG] Executable: "%JAN_EXE%"
echo.

REM === Check executable ===
if not exist "%JAN_EXE%" (
    echo [ERROR] Jan.exe not found at:
    echo         "%JAN_EXE%"
    pause
    exit /b 1
)

REM === Avoid double instance ===
tasklist | find /i "Jan.exe" >nul
if %errorlevel%==0 (
    echo [INFO] Jan is already running. Another instance will not be opened.
    exit /b 0
)

echo Launching Jan Portable...
start "" "%JAN_EXE%"

echo.
echo [OK] Jan Portable started successfully.
endlocal
exit /b 0



```




Folder structure:
```

JanPortable/
├── JanPortable.bat         ← The .bat file
└── AppData/
    ├── Local/
    │   └── jan.ai.app/        
    │       ├── data/
    │       └── EBWebView/
    └── Roaming/
        └── Jan/               
            ├── Jan.exe        
            ├── data/
            │   ├── assistants/
            │   ├── extensions/
            │   ├── llamacpp/
            │   │   ├── backends/
            │   │   ├── lib/
            │   │   └── models/
            │   ├── logs/
            │   └── threads/
            └── resources/
                └── pre-install/

```


PS: When Jan.ai is launched, for the first time:

got to Settings -> General -> scroll down to Advanced and click "Reset", close Jan and open it again using the .bat
